### PR TITLE
⌨️ style(a11y): kb access for LLM endpoint menu

### PR DESCRIPTION
**Pull request by berry-13**
## Summary

users can now select, open and choose endpoint with the keyboard

### dark
| before | after |
|--------|-------|
| image | image |
| image | image |
| image | image | 

### light

| before | after |
|--------|-------|
| image | image |
| image | image |
| image | image | 

## Change Type

- [x] Accessibility

### CODE REVIEW

**Comment by danny-avila** [comment]
thanks! it's addressed by https://github.com/danny-avila/LibreChat/pull/3536 so no action on this

File: client/src/style.css [hunk 1]
```diff
@@ -2205,6 +2205,29 @@ ol ol):not(:where([class~=not-prose] *)) {
   
 }
 
+.focus-visible-button {
+  background-color: transparent;
+  transition: background-color 0.2s;
+}
+
+* {
```
Comment (danny-avila) [inline client/src/style.css:2213]: is this right? every element should have this?

File: .gitignore [hunk 2]
```diff
@@ -97,6 +97,7 @@ data.ms/*
 auth.json
 
 /packages/ux-shared/
+/packages/data-provider/undefined/
```
Comment (danny-avila) [inline .gitignore:100]: `/undefined/` What's the purpose for this change?

File: client/src/components/Chat/Input/Files/AttachFile.tsx [hunk 3]
```diff
@@ -45,7 +45,7 @@ const AttachFile = ({
         <button
           disabled={!!disabled}
           type="button"
-          className="btn relative text-black focus:outline-none focus:ring-2 focus:ring-border-xheavy focus:ring-opacity-50 dark:text-white"
+          className="btn text-blac relative dark:text-white"
```
Comment (danny-avila) [inline client/src/components/Chat/Input/Files/AttachFile.tsx:48]: text-blac is not a valid TWCSS class

File: client/src/components/Chat/Input/HeaderOptions.tsx [hunk 4]
```diff
@@ -83,18 +83,20 @@ export default function HeaderOptions({
...
               {!noSettings[endpoint] && interfaceConfig?.parameters && (
-                <Button
+                <button
                   type="button"
+                  id="parameters-button"
+                  data-testid="parameters-button"
                   className={cn(
-                    cardStyle,
-                    'z-50 flex h-[40px] min-w-4 flex-none items-center justify-center px-3 focus:ring-0 focus:ring-offset-0',
+                    'pointer-cursor relative flex flex-col rounded-md border border-gray-100 bg-white text-left dark:border-gray-700 dark:bg-gray-800 sm:text-sm',
```
Comment (danny-avila) [inline client/src/components/Chat/Input/HeaderOptions.tsx:91]: Any time you touch elements with legacy colors, please apply theme classnames

File: client/src/components/Chat/Input/OptionsPopover.tsx [hunk 5]
```diff
@@ -66,7 +66,7 @@ export default function OptionsPopover({
               {presetsDisabled ? null : (
                 <Button
                   type="button"
-                  className="h-auto w-[150px] justify-start rounded-md border border-gray-300/50 bg-transparent px-2 py-1 text-xs font-medium font-normal text-black hover:bg-gray-100 hover:text-black focus:ring-1 focus:ring-green-500/90 dark:border-gray-500/50 dark:bg-transparent dark:text-white dark:hover:bg-gray-600 dark:focus:ring-white"
+                  className="h-auto w-[150px] justify-start rounded-md border border-gray-300/50 bg-transparent px-2 py-1 text-xs font-medium font-normal text-black hover:bg-gray-100 hover:text-black dark:border-gray-500/50 dark:bg-transparent dark:text-white dark:hover:bg-gray-600"
```
Comment (danny-avila) [inline client/src/components/Chat/Input/OptionsPopover.tsx:69]: Any time you touch elements with legacy colors, please apply theme classnames

File: client/src/components/Chat/Menus/Endpoints/MenuItem.tsx [hunk 6]
```diff
@@ -136,12 +134,11 @@ const MenuItem: FC<MenuItemProps> = ({
               <div className="text-token-text-primary" key={`set-key-${endpoint}`}>
                 <button
                   className={cn(
-                    'invisible flex gap-x-1 group-hover:visible',
+                    ' invisible flex gap-x-1 group-hover:visible',
                     selected ? 'visible' : '',
                     expiryTime
-                      ? 'w-full rounded-lg p-2 hover:text-gray-400 dark:hover:text-gray-400'
+                      ? 'w-full rounded-lg p-2 text-gray-400 hover:text-black dark:text-gray-400 dark:hover:text-white'
```
Comment (danny-avila) [inline client/src/components/Chat/Menus/Endpoints/MenuItem.tsx:140]: Any time you touch elements with legacy colors, please apply theme classnames, especially don't add legacy aka `*-gray-*`

File: client/src/components/Chat/Menus/PresetsMenu.tsx [hunk 7]
```diff
@@ -27,7 +27,7 @@ const PresetsMenu: FC = () => {
       <Trigger asChild>
         <button
           className={cn(
-            'pointer-cursor relative flex flex-col rounded-md border border-gray-100 bg-white text-left focus:ring-0 focus:ring-offset-0 dark:border-gray-700 dark:bg-gray-800 sm:text-sm',
+            'pointer-cursor relative flex flex-col rounded-md border border-gray-100 bg-white text-left dark:border-gray-700 dark:bg-gray-800 sm:text-sm',
```
Comment (danny-avila) [inline client/src/components/Chat/Menus/PresetsMenu.tsx:30]: Any time you touch elements with legacy colors, please apply theme classnames, especially don't add legacy aka `*-gray-*`

File: client/src/components/Chat/Menus/UI/TitleButton.tsx [hunk 8]
```diff
@@ -1,12 +1,19 @@
...
 
 export default function TitleButton({ primaryText = '', secondaryText = '' }) {
+  const [isExpanded, setIsExpanded] = useState(false);
+
   return (
     <Trigger asChild>
       <button
-        className="group flex cursor-pointer items-center gap-1 rounded-xl px-3 py-2 text-lg font-medium hover:bg-gray-50 radix-state-open:bg-gray-50 dark:hover:bg-gray-700 dark:radix-state-open:bg-gray-700"
-        type="button"
+        className=" group relative flex cursor-pointer items-center gap-1 rounded-xl px-3 py-2 text-lg font-medium hover:bg-gray-100 radix-state-open:bg-gray-100 dark:hover:bg-gray-700 dark:radix-state-open:bg-gray-700"
```
Comment (danny-avila) [inline client/src/components/Chat/Menus/UI/TitleButton.tsx:10]: Any time you touch elements with legacy colors, please apply theme classnames, especially don't add legacy aka `*-gray-*`

File: client/src/components/Plugins/Store/PluginStoreDialog.tsx [hunk 9]
```diff
@@ -187,7 +187,7 @@ function PluginStoreDialog({ isOpen, setIsOpen }: TPluginStoreDialogProps) {
...
                     onChange={handleSearch}
                     placeholder={localize('com_nav_plugin_search')}
-                    className="flex rounded-md border border-gray-200 bg-transparent py-2 pl-10 pr-2 shadow-[0_0_10px_rgba(0,0,0,0.05)] outline-none placeholder:text-gray-400 focus:border-gray-400 focus:bg-gray-50 focus:outline-none focus:ring-0 focus:ring-gray-400 focus:ring-opacity-0 focus:ring-offset-0 disabled:cursor-not-allowed disabled:opacity-50 dark:border-gray-600 dark:bg-gray-700 dark:text-gray-50 dark:shadow-[0_0_15px_rgba(0,0,0,0.10)] dark:focus:border-gray-500 focus:dark:bg-gray-600 dark:focus:outline-none dark:focus:ring-0 dark:focus:ring-gray-500 dark:focus:ring-offset-0 dark:focus:ring-offset-gray-900"
+                    className="flex rounded-md border border-gray-200 bg-transparent py-2 pl-10 pr-2 shadow-[0_0_10px_rgba(0,0,0,0.05)] outline-none placeholder:text-gray-400 focus:border-gray-400 focus:bg-gray-50 focus:outline-none focus:ring-0 focus:ring-gray-400 focus:ring-opacity-0 focus:ring-offset-0 disabled:cursor-not-allowed disabled:opacity-50 dark:border-gray-600 dark:bg-gray-700 dark:text-gray-50 dark:shadow-[0_0_15px_rgba(0,0,0,0.10)]"
```
Comment (danny-avila) [inline client/src/components/Plugins/Store/PluginStoreDialog.tsx:190]: Any time you touch elements with legacy colors, please apply theme classnames, especially don't add legacy aka `*-gray-*`

File: client/src/components/ui/Button.tsx [hunk 10]
```diff
@@ -3,7 +3,7 @@ import { VariantProps, cva } from 'class-variance-authority';
 import { cn } from '~/utils';
 
 const buttonVariants = cva(
-  'rounded-md inline-flex items-center justify-center text-sm font-medium transition-colors dark:hover:bg-gray-700 dark:hover:text-gray-100 disabled:opacity-50 disabled:pointer-events-none data-[state=open]:bg-gray-100 dark:data-[state=open]:bg-gray-700 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-gray-500',
+  'rounded-md inline-flex items-center justify-center text-sm font-medium transition-colors dark:hover:bg-gray-700 dark:hover:text-gray-100 disabled:opacity-50 disabled:pointer-events-none data-[state=open]:bg-gray-100 dark:data-[state=open]:bg-gray-700',
```
Comment (danny-avila) [inline client/src/components/ui/Button.tsx:6]: Any time you touch elements with legacy colors, please apply theme classnames, especially don't add legacy aka `*-gray-*`

File: client/src/components/ui/Combobox.tsx [hunk 11]
```diff
@@ -72,7 +72,7 @@ export default function ComboboxComponent({
             isCollapsed
               ? 'flex h-9 w-9 shrink-0 items-center justify-center p-0 [&>span]:w-auto [&>svg]:hidden'
               : '',
-            'bg-white text-black hover:bg-gray-50 focus-visible:ring-2 focus-visible:ring-gray-500 dark:bg-gray-850 dark:text-white ',
+            'bg-white text-black hover:bg-gray-50 dark:bg-gray-850 dark:text-white ',
```
Comment (danny-avila) [inline client/src/components/ui/Combobox.tsx:75]: Any time you touch elements with legacy colors, please apply theme classnames, especially don't add legacy aka `*-gray-*`

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: client/src/style.css (hunk 1 above)
Comment (danny-avila) [reply client/src/style.css:2213]: Applying a transition to every element using the universal selector (*) is generally considered bad practice because:

1. It can significantly impact performance, especially on large websites.
2. It may cause unexpected and unwanted transitions on elements.
3. It complicates CSS specificity and makes overriding styles more difficult.
4. It can lead to maintenance issues and make debugging more challenging.
5. It might create accessibility problems for users sensitive to motion.

Instead, it's better to apply transitions selectively to specific elements or classes where they are actually needed.

Comment (berry-13) [reply client/src/style.css:2213]: yes, I think that adding a transition color to all the elements will reduce the className lenght and make everything smoother without adding `transition-color` and `duration-200`

File: .gitignore (hunk 2 above)
Comment (berry-13) [reply .gitignore:100]: sometimes when building the data-provider's packages, some random files were created under this directory

Comment (danny-avila) [reply .gitignore:100]: yes I've noticed but they usually get cleared, I'll accept this then

**Comment by danny-avila** [comment]
Not super high priority but I think you understand this better now: 

> Any time you touch elements with legacy colors, please apply theme classnames, especially don't add legacy aka `*-gray-*`

so please resolve the comments I made when you have the time

File: client/src/components/ui/Combobox.tsx (hunk 11 above)
Comment (berry-13) [reply client/src/components/ui/Combobox.tsx:75]: @danny-avila am I wrong or this component isn't used anymore?
